cli.js

```diff
@@ -129,7 +129,7 @@ var api = new Api({
 
 var reporter;
 
-if (cli.flags.tap) {
+if (cli.flags.tap && !cli.flags.watch) {
```

Comment:
If they do `--tap --watch` it should throw. ("watch is not allowed with tap reporter")

If they do `--watch` and their pkgConf specifies `tap:true`, then it should fall through to picking `verbose` or `mini` reporter.